jsonrpcserver/dispatcher.py

```diff
@@ -132,6 +136,13 @@ def handle_exceptions(request: Request, debug: bool) -> Generator:
         handler.response = InvalidParamsResponse(
             id=request.id, data=str(exc), debug=debug
         )
+    except ApiError as exc: # Method signals custom error
+        handler.response = ErrorResponse(
+                str(exc), code=exc.code, data=exc.data,
+                id=request.id,
+                # always set debug to send data to client
+                debug=True,
```

Comment:
Can we make debug a property of the exception so the user can control this?
```python
raise ApiError("Invalid file", data=filename, debug=True)
```

Reply:
I think, this does not make much sense because for `ApiError`s the `data` is provided by the method and part of the method's API. So, you always want to send `data` in that case.
However, globally setting `debug=True` would send internal data if a method fails because of programming errors etc., too, which is clearly not what is wanted in production code.
Maybe, the parameter `debug` of `ErrorResponse` should be renamed to `include_data` to make the intention more clear.

Reply:
I guess you're right. When raising an ApiError you're expecting the information to be returned to the client. I can't see any reason to exclude the data part.